include/quicr/message_buffer.h

```diff
@@ -56,6 +56,27 @@ namespace quicr::messages {
  */
 class MessageBuffer
 {
+public:  
+  struct ReadException : public std::runtime_error
+  {
+    using std::runtime_error::runtime_error;
+  };
+
+  struct OutOfRangeException : public ReadException
```

Comment:
While this may be correct for now, won't we have write exceptions too? Matching on the name ```ReadException``` might be misleading when there could be an ```OutOfRangeException``` for write operations.  Seems like ```ReadException``` could be renamed to ```BaseException``` or ```GeneralException```

Reply:
Yeah my brain kind of went auto-pilot and called it Read exception. In all honesty since the usage is mainly `MessageBuffer::ReadException` maybe this could even simply be called `Exception`

I don't think we have any write exceptions. Should we? What kind of exceptions could happen on write?

Reply:
If ```read == Out == Decode``` then we have one **write** on line https://github.com/Quicr/new-libquicr/blob/main/src/message_buffer.cpp#L229.   

I think Read/Write is misleading.  To me, **In / Out** might be more intuitive.  For example, all exceptions that are involved with adding data to a message buffer would be considered **In**. 

A use-case, which we aren't using today... yet, is that we might want to handle/catch all ```InExceptions``` differently than ```OutExceptions```.  I suppose we can add that later (simple header change) if we used ```Exception``` for now.

Reply:
That exception was removed in this PR since the assertion is done in the constructor of `uintVar_t` now, so there shouldn't be any write exceptions anymore.